packages/core/src/new-api/internal/deployment-loader/file-deployment-loader.ts

```diff
@@ -0,0 +1,131 @@
+import fs from "fs-extra";
+import path from "path";
+
+import { Artifact, BuildInfo } from "../../types/artifact";
+import { DeploymentLoader } from "../../types/deployment-loader";
+import { Journal } from "../../types/journal";
+import { FileJournal } from "../journal/file-journal";
+import { MemoryJournal } from "../journal/memory-journal";
+import { assertIgnitionInvariant } from "../utils/assertions";
+
+export class FileDeploymentLoader implements DeploymentLoader {
+  public journal: Journal;
+
+  private _paths: {
+    deploymentDir: string;
+    artifactsDir: string;
+    buildInfoDir: string;
+    journalPath: string;
+    deployedAddressesPath: string;
+  } | null = null;
+
+  constructor(private readonly _deploymentDirPath: string) {
+    this.journal = new MemoryJournal();
+  }
+
+  public async initialize(): Promise<void> {
```

Comment:
I think it would be better to make this private and call `await initialize()` at the start of every public method.

This requires turning `initialize()` into a no-op if run twice.

Reply:
I made `initialize` private and idempotent.